src/Tribe/iCal.php

```diff
@@ -157,14 +157,89 @@ public function do_ical_template() {
 				$events    = tribe_get_events( array( 'post__in' => $event_ids ) );
 				$this->generate_ical_feed( $events );
 			} elseif ( is_single() ) {
-				$this->generate_ical_feed( $wp_query->post );
+				if ( is_singular( Tribe__Events__Main::POSTTYPE ) ) {
+					// If the single view is rendered we need to render the post only if is an Event
+					$this->generate_ical_feed( $wp_query->post );
+				} else {
+					// otherwise it might be a shortcode.
+					$this->generate_ical_feed( $this->find_events_in_shortcode() );
```

Comment:
This looks like a solid working solution, _but,_ I'm a little uncomfortable with us adding special handling for a PRO feature inside of TEC. As a rule of thumb, the dependency (PRO) should have knowledge of the parent (TEC) but not the other way round.

Is it possible for us to approach this by making the list of post IDs used to generate the ical feed filterable, and filtering it from within the shortcode logic contained in PRO?

(cc @bordoni for any thoughts on this)

Reply:
Got it, so we just need to update as follows: 

1. Add a filter to filter list of events IDs.
2. Add a callback on PRO about this change
3. Move the logic all the way there (in PRO).

Let me know what do you think.

Thanks

Reply:
Yes! Adding a filter would anyway be useful outside of this scenario, so it feels like a great way to go :+1: